api/v1/pooler_types.go

```diff
@@ -126,6 +130,20 @@ type PodTemplateSpec struct {
 	Spec corev1.PodSpec `json:"spec,omitempty"`
 }
 
+// ServiceTemplateSpec is a structure allowing the user to set
+// a template for Service generation.
+type ServiceTemplateSpec struct {
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	ObjectMeta metav1.ObjectMeta `json:"metadata,omitempty"`
```

Comment:
```suggestion
	ObjectMeta Metadata `json:"metadata,omitempty"`
```
Same as the pod template above, will need to be converted